### PR TITLE
typing changes for activate

### PR DESCRIPTION
**Pull request by dholth**
### Description

Add typing to activate, to help with updating test_activate.py

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: conda/activate.py [hunk 1]
```diff
@@ -410,23 +429,25 @@ def build_deactivate(self):
                 conda_default_env=None,
                 conda_prompt_modifier=None,
             )
-            conda_prompt_modifier = ''
+            conda_prompt_modifier = ""
             activate_scripts = ()
-            export_path = {'PATH': new_path, }
+            export_path = {
+                "PATH": new_path,
+            }
```
Comment (kenodegard) [inline conda/activate.py:436]:
```suggestion
            export_path = {"PATH": new_path}
```

File: conda/activate.py [hunk 2]
```diff
@@ -788,84 +816,87 @@ def path_identity(paths):
...
         # Best solution appears to be https://stackoverflow.com/a/1250279
         ps1 = ps1.replace("'", "'\"'\"'")
-        set_vars.update({
-            'PS1': conda_prompt_modifier + ps1,
-        })
+        set_vars.update(
+            {
+                "PS1": conda_prompt_modifier + ps1,
+            }
+        )
```
Comment (kenodegard) [inline conda/activate.py:852]:
```suggestion
        set_vars["PS1"] = conda_prompt_modifier + ps1
```

File: conda/activate.py [hunk 3]
```diff
@@ -788,84 +816,87 @@ def path_identity(paths):
...
-            prompt = re.sub(re.escape(current_prompt_modifier), r'', prompt)
-        set_vars.update({
-            'prompt': conda_prompt_modifier + prompt,
-        })
+            prompt = re.sub(re.escape(current_prompt_modifier), r"", prompt)
+        set_vars.update(
+            {
+                "prompt": conda_prompt_modifier + prompt,
+            }
+        )
```
Comment (kenodegard) [inline conda/activate.py:899]:
```suggestion
        set_vars["prompt"] = conda_prompt_modifier + prompt
```

**Review by kenodegard — CHANGES_REQUESTED** [review]
Minor edits

File: conda/activate.py (hunk 2 above)
Comment (dholth) [reply conda/activate.py:852]: I ran this file through black to silence the linter. I would rather avoid making other stylistic changes.

**Review by kenodegard — APPROVED** [review]

**Pull request merged**